test/functional/terminal/tui_spec.lua

```diff
@@ -223,6 +237,21 @@ describe('TUI :restart', function()
     -- Check ":restart!".
     tt.feed_data(':restart!\013')
     screen:expect(s0)
+    restart_pid_check()
+
+    tt.feed_data(':echo\n')
+    screen:expect([[
+      ^                                                  |
+      {4:~                                                 }|*3
+      {5:[No Name]                                         }|
+                                                        |
+      {3:-- TERMINAL --}                                    |
+    ]])
+
+    -- :restart also works without --listen conflict when server exit is delayed.
```

Comment:
Took me a second to understand this
```suggestion
    -- :restart avoids a --listen conflict when server exit is delayed.
```